### PR TITLE
Upgrade typescript

### PR DESCRIPTION
**Pull request by rgomezp**
# Description
## 1 Line Summary
Upgrades Typescript version to 3.9.7

# Systems Affected
* [x] WebSDK
* [ ] Backend
* [ ] Dashboard

# Validation
## Tests
All tests pass

### Info
Fixed Unit Tests that broke due to Typescript changes

### Checklist
* [x] All the automated tests pass or I explained why that is not possible
* [x] I have personally tested this on my machine or explained why that is not possible
* [ ] I have included test coverage for these changes or explained why they are not needed

**Programming Checklist**
Interfaces:

* [x] Don't use default export
* [x] New interfaces are in model files
  * New interfaces are unique to each individual testing file

Functions:

* [x] Don't use default export
* [x] All function signatures have return types
* [x] Helpers should not access any data but rather be given the data to operate on.

Typescript:

* [x] No Typescript warnings
* [x] Avoid silencing null/undefined warnings with the exclamation point

Other:

* [x] Iteration: refrain from using `elem of array` syntax. Prefer `forEach` or use `map`
* [x] Avoid using global OneSignal accessor for `context` if possible. Instead, we can pass it to function/constructor so that we don't call `OneSignal.context`

### Info
No relevant screenshots necessary

### Checklist
* [x] I have included screenshots/recordings of the intended results or explained why they are not needed

- - -
## Related Tickets
- - -
## 
This change is [Reviewable](https://reviewable.io/reviews/onesignal/onesignal-website-sdk/727)

### CODE REVIEW

File: src/utils.ts [hunk 1]
```diff
@@ -379,11 +379,9 @@ export function getSdkLoadCount() {
 export async function awaitSdkEvent(eventName: string, predicate?: Action<any>) {
   return await new Promise(resolve => {
     OneSignal.emitter.once(eventName, (event: Event) => {
-      if (predicate) {
-        const predicateResult = predicate(event);
-        if (predicateResult)
-          resolve(event);
-      } else resolve(event);
+      if (!predicate) {
```
Comment (jmadler) [inline src/utils.ts:382]: This seems like a meaningful semantic shift -- what's the intent here?

Comment (rgomezp) [reply src/utils.ts:382]: See the commit for [more details](https://github.com/OneSignal/OneSignal-Website-SDK/commit/ed16f5710b57a268401ae29ae59b1f4bca08cf3b)

It looks like `predicateResult` is always of type `void` anyway.

Comment (jmadler) [reply src/utils.ts:382]: Makes sense, but it looks like we're no longer called predicate when present.  Is that avoidable?

Comment (rgomezp) [reply src/utils.ts:382]: Decided on removing predicate altogether....

```
export async function awaitSdkEvent(eventName: string) {
  return await new Promise(resolve => {
    OneSignal.emitter.once(eventName, (event: Event) => {
      resolve(event);
    });
  });
}
```

This function is not used very much and it never actually receives a `predicate`.

**Comment by rgomezp** [comment]
Closing in favor of cleaned up PR:  https://github.com/OneSignal/OneSignal-Website-SDK/pull/735

**Pull request closed without merging**